### PR TITLE
Fix a bug where the filename variable can get clobbered due to XmlReader invalidating the buffer

### PR DESCRIPTION
**Pull request by JesseCol**
The buffer returned by IXmlReader::GetValue gets invalidated after the XmlReader is used more in the future.  The fileName buffer was getting re-used in a scenario where there were lots of nodes to read.  The fix is simply to save it off to a local wstring.

### CODE REVIEW

File: src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/dllmain.cpp [hunk 1]
```diff
@@ -442,10 +442,12 @@ HRESULT ExtRoLoadCatalog()
 
 BOOL WINAPI DllMain(HINSTANCE hmodule, DWORD reason, LPVOID /*lpvReserved*/)
 {
+    /*
```
Comment (kythant) [inline src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/dllmain.cpp:445]: What's the reason for this change?

**Comment by ocalvo** [comment]
The commend in dllmain.cpp was added to assist testing the change. It should had been removed.
I have created a new PR based on thos one here: https://github.com/microsoft/xlang/pull/767

**Comment by JesseCol** [comment]
> The commend in dllmain.cpp was added to assist testing the change. It should had been removed. I have created a new PR based on this one [here](https://github.com/microsoft/xlang/pull/767)

Yep, sorry, my bad.  I'll close this PR out since Oscar created a new one.
